packages/location-field/src/index.tsx

```diff
@@ -0,0 +1,758 @@
+    setMenuVisible(true);
+    if (nearbyStops.length === 0 && currentPosition && currentPosition.coords) {
+      findNearbyStops({
+        lat: currentPosition.coords.latitude,
+        lon: currentPosition.coords.longitude,
+        max: geocoderConfig.maxNearbyStops || 4
+      });
+    }
+  };
+
+  const onKeyDown = evt => {
+    switch (evt.key) {
+      // 'Down' arrow key pressed: move selected menu item down by one position
+      case "ArrowDown":
+        // Suppress default 'ArrowDown' behavior which moves cursor to end
+        evt.preventDefault();
+        if (!menuVisible) {
+          // If the menu is not visible, simulate a text input click to show it.
+          handleTextInputClick();
+        } else if (activeIndex === menuItemCount - 1) {
+          setActiveIndex(null);
+        } else {
+          setActiveIndex(activeIndex === null ? 0 : activeIndex + 1);
+        }
+        break;
+      // 'Up' arrow key pressed: move selection up by one position
+      case "ArrowUp":
+        // Suppress default 'ArrowUp' behavior which moves cursor to beginning
+        evt.preventDefault();
+        if (activeIndex === 0) {
+          setActiveIndex(null);
+        } else {
+          setActiveIndex(
+            activeIndex === null ? menuItemCount - 1 : activeIndex - 1
+          );
+        }
+        break;
+      // 'Enter' keypress serves two purposes:
+      //  - If pressed when typing in search string, switch from 'autocomplete'
+      //    to 'search' geocoding
+      //  - If pressed when dropdown results menu is active, apply the location
+      //    associated with current selected menu item
+      case "Enter":
+        if (typeof activeIndex === "number") {
+          // Menu is active
+          // Retrieve location selection handler from lookup object and invoke
+          const locationSelected = locationSelectedLookup[activeIndex];
+          if (locationSelected) locationSelected();
+
+          // Clear selection & hide the menu
+          setMenuVisible(false);
+          setActiveIndex(null);
+        } else {
+          // Menu not active; get geocode 'search' results
+          geocodeSearch(evt.target.value);
+          // Ensure menu is visible.
+          setMenuVisible(true);
+        }
+
+        // Suppress default 'Enter' behavior which causes page to reload
+        evt.preventDefault();
+        break;
+      case "Escape":
+      case "Tab":
+        // Clear selection & hide the menu
+        setMenuVisible(false);
+        setActiveIndex(null);
+        break;
+      // Any other key pressed: clear active selection
+      default:
+        setActiveIndex(null);
+        break;
+    }
+  };
+
+  const geocodeSearch = text => {
+    if (!text) {
+      console.warn("No text entry provided for geocode search.");
+      return;
+    }
+    getGeocoder(geocoderConfig)
+      .search({ text })
+      .then(result => {
+        if (result.features && result.features.length > 0) {
```

Comment:
Use `?.` notation.

Reply:
Addressed in 18d5606109e7dfce7d00d42b42c8b1b6f075a0bc